src/secret_sharing/mod.rs

```diff
@@ -48,6 +51,16 @@ pub trait Block: Sized + Copy + Debug {
     type Size: ArrayLength<u8>;
 }
 
+pub trait WeakSharedValue:
```

Comment:
just to debate it a bit - what would be a good name for this type? does it look like a group element under addition?

Reply:
dont have a strong opinion, we have SharedValue that also implements mult.. For a group you technically need that you have inverse elements (-) and a neutral element (0). It seems that this is the case for all our use cases. However, would we call SharedValue then a ring or field element?

Reply:
well, neutral element is here (ZERO). Negate is here too, SharedValue could be ring and field can get `inv` if we need to :) I can help if we all agree that this is what needs to be done - I don't think we should make it part of this PR though.